### PR TITLE
Refactored - FindElement/s and Server Stop 

### PR DESCRIPTION
**Pull request by sravanmedarapu**
Refactored following

-  Server Stop(DELETE SESSION)
- FindElement/s getXPathUiObjects
- GetElementAttribute Handler
- FindElements to find elements with contextId
- Clear & SendKeys to perform on focused element on absence of elementId

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/handler/GetElementAttribute.java [hunk 1]
```diff
@@ -27,13 +30,30 @@ public AppiumResponse safeHandle(IHttpRequest request) {
         if (element == null) {
             return new AppiumResponse(getSessionId(request), WDStatus.NO_SUCH_ELEMENT, "Element Not found");
         }
-        Object text = JSONObject.NULL;
         try {
-            text = element.getStringAttribute(attributeName);
+            if (attributeName.equals("name") || attributeName.equals("contentDescription")
```
Comment (mpujari) [inline app/src/main/java/io/appium/uiautomator2/handler/GetElementAttribute.java]: Minor: Can we change the usage to check equals as below

```
"name".equals(attributeName) || "contentDescription".equals(attributeName)....
```

File: app/src/main/java/io/appium/uiautomator2/model/UiObject2Element.java [hunk 2]
```diff
@@ -56,7 +57,36 @@ public String getStringAttribute(final String attr) throws UiObjectNotFoundExcep
...
-            throw new NoSuchElementAttributeException("The attribute with name '" + attr + "' was not found.");
+            throw new NoAttributeFoundException(attr);
+        }
+        return res;
+    }
+
+    public boolean getBoolAttribute(final String attr)
+            throws UiObjectNotFoundException, NoAttributeFoundException, UiAutomator2Exception {
+        boolean res;
+        if (attr.equals("enabled")) {
```
Comment (mpujari) [inline app/src/main/java/io/appium/uiautomator2/model/UiObject2Element.java]: Same as above, can we have "enabled".equals(attr) and where ever equals is called.

File: app/src/main/java/io/appium/uiautomator2/handler/FindElements.java [hunk 3]
```diff
@@ -157,38 +157,31 @@ public AppiumResponse safeHandle(IHttpRequest request) {
         if (element == null) {
             throw new ElementNotFoundException();
         }
-        if (by instanceof By.ById) {
-            String locator = getElementLocator((ById) by);
-            return getInstance().findObjects(android.support.test.uiautomator.By.res(locator));
+        if (by instanceof ById) {
+            String locator = getElementLocator((ById)by);
+            return element.getChilds(android.support.test.uiautomator.By.res(locator));
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/FindElements.java]: Is `getChilds` actually the name of the method? Do we have control over that? If we do, that should be `getChildren`.

File: app/src/main/java/io/appium/uiautomator2/model/AndroidElement.java [hunk 4]
```diff
@@ -29,6 +36,8 @@
 
     public Object getChild(final Object sel) throws UiObjectNotFoundException, InvalidSelectorException, ClassNotFoundException;
 
+    public List<Object> getChilds(final Object selector) throws UiObjectNotFoundException, InvalidSelectorException, ClassNotFoundException;
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/model/AndroidElement.java]: The plural of "child" is "children".

File: app/src/main/java/io/appium/uiautomator2/utils/XMLHierarchy.java [hunk 5]
```diff
@@ -29,16 +29,55 @@
...
+        XPathExpression exp = null;
+        try {
+            exp = xpath.compile(xpathExpression);
+        } catch (XPathExpressionException e) {
+            throw new InvalidSelectorException("Invalid XPath expression: ", e);
+        }
+        return exp;
+    }
+
+   /* public static ArrayList<ClassInstancePair> getClassInstancePairs(XPathExpression xpathExpression, Node root) throws ElementNotFoundException {
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/utils/XMLHierarchy.java]: Why is this commented out? It ought to either be commented to make clear why it is here at all, or deleted full stop.

**Comment by imurchie** [comment]
Other than "getChilds", this looks good from a cursory glance.

**Pull request merged**